tests/component/views/create-dataset/CreateDataset.spec.tsx

```diff
@@ -0,0 +1,24 @@
+import { CreateDataset } from '../../../../src/views/create-dataset/CreateDataset'
+
+describe('Form component', () => {
+  it('should render all form fields', () => {
+    cy.customMount(<CreateDataset />)
+    cy.get('[data-cy=datasetFormInputTitle]').should('be.visible')
+    cy.get('[data-cy=datasetFormSubmit]').should('be.visible')
+    cy.get('[data-cy=datasetFormCancel]').should('be.visible')
+  })
+
+  it('should submit the form with correct values', () => {
+    cy.customMount(<CreateDataset />)
+
+    cy.get('[data-cy=datasetFormInputTitle]').type('Test Dataset Title')
+
+    cy.get('[data-cy=datasetFormSubmit]').click()
+
+    // Wait for the form submission to complete (adjust the timeout as needed)
+    cy.wait(4000)
+
+    // Assert that the form was successfully submitted
+    cy.contains('Form Submitted!')
```

Comment:
I think you don't need an wait here, if something is going to appear after some time then there is no need for a cy.wait, since Cypress will eventually find the` Form submitted` text as long as it does not surpass the cypress asserts timeout which it shouldn't

Reply:
OK, great news. I added an action timeout since I think the default timeout is less than the wait time for the mock api call I was using.